### PR TITLE
Remove VestibuleSlice.js and cleanup some redux state everywhere.

### PR DESCRIPTION
**Pull request by xmquentin**
Ah state looks good! Not perfect, but pretty clean, and the code base feels a little more easy to understand IMO.

This works for:
* Creating & entering strict/non-strict rooms
* Joining & entering strict/non-strict rooms with 2+ people
* Chatroom messages

When doing chat messages, the RAS actually lost connection with the frontend somehow. That is to be looked into (cough hridayesh@)

Not going to post a gif, going to have to trust me or try it out for yourself. <3

### CODE REVIEW

File: src/components/Vestibule/VestibulePanel/VestibulePanel.js [hunk 1]
```diff
@@ -146,7 +140,7 @@ function VestibulePanel() {
    * @function getRoomUserNameChar - get first char of Room UserName
    */
    function getRoomUserNameChar() {
-    return store.getState().room.roomUserName.substring(0, 1);
+    return store.getState().room.roomUserName !== null ? store.getState().room.roomUserName.substring(0, 1) : "?";
```
Comment (ABFranco) [inline src/components/Vestibule/VestibulePanel/VestibulePanel.js:143]: nice

File: src/components/Vestibule/VestibulePanel/VestibulePanel.js [hunk 2]
```diff
@@ -6,15 +6,14 @@ import Avatar from '@material-ui/core/Avatar';
...
+import { clearRoomData, roomJoinCancel, roomDelete, roomLeave, setEnteredRoom } from '../../../reducers/RoomSlice';
 import { clearChatHistory } from '../../../reducers/ChatroomSlice';
-import { clearVestibuleData } from '../../../reducers/VestibuleSlice';
 import { setErrorMessage } from '../../../reducers/NotificationSlice';
 import store from '../../../store';
 
 function VestibulePanel() {
   const dispatch = useDispatch();
-  const isWaiting = useSelector(state => (state.vestibule.roomId && state.vestibule.roomPassword && state.vestibule.userName));
+  const isWaiting = useSelector(state => (!state.room.userIsHost && !state.room.roomJoinRequestAccepted));
```
Comment (ABFranco) [inline src/components/Vestibule/VestibulePanel/VestibulePanel.js:16]: I see

File: src/components/Room/Room.js [hunk 3]
```diff
@@ -55,11 +53,14 @@ function Room() {
...
     if (userInRoom && !store.getState().room.userIsHost && store.getState().room.token !== null) {
       joinRoomClosureStream();
     }
+    // re-join media room upon refresh if in the room
     if (userInRoom && store.getState().room.token !== null) {
-      joinMediaRoom();
+      // TODO: test media room functionality.
+      // joinMediaRoom();
     }
   },[userInRoom]);
```
Comment (ABFranco) [inline src/components/Room/Room.js]: We might need to look more into when `joinRoomClosureStream` is called. If the user is in the vestibule, they need to know that the Room is closed.
Or maybe we need to do some sort of backend call to "check if room is closed" b/c RFE still allows the user to visually enter the Room even if it got closed by the host. This is a minor edge case so I'm unsure how much it is worth investigating.

File: src/components/Room/Room.js [hunk 4]
```diff
@@ -496,42 +487,47 @@ function Room() {
...
+            userId: chatHistory[i].getUserId(),
+            name: chatHistory[i].getUserName(),
+            message: chatHistory[i].getMessage(),
+            timestamp: chatHistory[i].getTimestamp(),
+          });
+        }
+        dispatch(setToken(token));
+        dispatch(setRoomJoinRequestAccepted(true));
+        dispatch(setChatHistory(chatHistoryData));
+        console.log('r.receiveJoinRoomResponse: userId=%o is ready to enter roomId=%o', store.getState().user.userId, roomId)
```
Comment (ABFranco) [inline src/components/Room/Room.js]: might be minor typo in the loggers with `r.` or just ignore b/c these don't really matter and can be deleted.

File: src/reducers/RoomSlice.js [hunk 5]
```diff
@@ -70,9 +73,22 @@ const roomSlice = createSlice({
     isStrict: false, 
     userIsHost: false,
     userInRoom: false,
-    roomUserName: null
+    userInVestibule: false,
+    roomUserName: null,
+    roomPassword: null,
+    roomJoinRequestAccepted: false,
```
Comment (ABFranco) [inline src/reducers/RoomSlice.js:79]: `roomJoinRequestAccepted` is fine to add, we could discuss later what this is supposed to represent for host vs non-host

**Review by ABFranco — APPROVED** [review]
Code looks fine and I did some testing on your branch and it looks good. Seems more stable than before and code did get some cleanup by removing the redundant state in the `vestibuleSlice`. We could build off these changes if we notice more bugs.

File: src/components/Room/Room.js (hunk 3 above)
Comment (xmquentin) [reply src/components/Room/Room.js]: Good catch. I will add a TODO to check this edge-case.

File: src/components/Room/Room.js (hunk 4 above)
Comment (xmquentin) [reply src/components/Room/Room.js]: oh I thought I saw you do this elsewhere. I'll remove.

**Comment by xmquentin** [comment]
@ABFranco Thx for actually checking out the branch

**Comment by xmquentin** [comment]
Got the LGTM via messenger from @hridayeshvjoshi

**Pull request merged**